### PR TITLE
fix(config): tolerate duplicate instance target names

### PR DESCRIPTION
**Pull request by EtienneLescot**
## Summary
- Allow existing v4 configs to load when instance target display names collide, while keeping target IDs strictly unique.
- Reuse or auto-name managed instance targets to avoid creating new name collisions from VS Code or CLI flows.
- Add config-service coverage for duplicate target names and managed/connected target collision handling.

## Validation
- npx vitest run tests/unit/config-service.test.ts
- npm run compile (packages/vscode-extension)

## Summary by CodeRabbit

* **Refactor**
  * Centralized creation/reuse of managed-local environment targets for consistent behavior across CLI and VS Code extension.
* **Bug Fixes**
  * Enforced unique instance-target identifiers and tightened environment API-key selection to avoid ambiguous/incorrect credential resolution.
* **Chores**
  * Persisted managed instance API keys into the manager secret store and expanded legacy API-key lookup locations.
* **Tests**
  * Added unit tests covering target reuse, name collisions, and credential-selection edge cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/EtienneLescot/n8n-as-code/pull/448?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: packages/cli/src/services/config-service.ts [hunk 1]
```diff
@@ -1819,7 +1844,7 @@ export class ConfigService {
         const rawEnvironments = raw.environments as unknown[];
         const environmentTargets = rawInstanceTargets.map((target, index) => this.sanitizeInstanceTarget(target, index));
         const environments = rawEnvironments.map((environment, index) => this.sanitizeEnvironment(environment, index));
-        this.assertUniqueIdsAndNames(environmentTargets, 'instance target');
+        this.assertUniqueIds(environmentTargets, 'instance target');
```
[the remark this answers is not in the document]
Comment (EtienneLescot) [reply packages/cli/src/services/config-service.ts:1866]: Handled in the follow-up commit on this PR. `readEnvApiKey` and `readTargetEnvApiKey` now only accept the name-based `N8NAC_TARGET_<target.name>_API_KEY` fallback when `isUniqueInstanceTargetName(target)` confirms the target name is unambiguous. When duplicate target names exist, only the ID-based env var is considered, so a connected target key can no longer override a managed target by name. Covered by `ignores name-based target API key variables when instance target names are ambiguous` in `config-service.test.ts`.
